app/src/main/java/com/amaze/filemanager/utils/ServiceWatcherUtil.java

```diff
@@ -192,4 +213,24 @@ public void run() {
 
         handler.postDelayed(runnable, 0);
     }
+
+    public interface ServiceWatcherInteractionInterface {
+
+        enum STATE {
+            HALTED,
+            RESUMED
+        }
+
+        /**
+         * Progress has been halted for some reason
+         */
+        void progressHalted();
+
+        /**
+         * Future extension for possible implementation of pause/resume of services
+         */
+        void progressResumed();
+
+        <T extends Service> T getServiceType();
```

Comment:
This method is redundant because of type erasure, just use the instance you are calling the method from (unless you can't access the Service instances, then you can have something like `Service getAttachedService();`).